### PR TITLE
Group ACL support

### PR DESCRIPTION
**Pull request by dmitrizagidulin**
Corresponds with solid-server PR https://github.com/solid/node-solid-server/pull/719

### CODE REVIEW

**Comment by kjetilk** [comment]
I can't get group ACLs working yet: https://pastebin.com/92cB4BB8

**Comment by kjetilk** [comment]
Here's my group listing:
```turtle
@prefix acl: <http://www.w3.org/ns/auth/acl#>.
@prefix vcard: <http://www.w3.org/2006/vcard/ns#> .

<> a acl:GroupListing.

<#meandi>
    a vcard:Group;
    vcard:hasUID <urn:uuid:8831CBAD-1111-2222-8563-F0F4787EG398:ABGroup> ;
    
    

  # Simply local and remote identity for myself
  vcard:hasMember </profile/card#me>;
  vcard:hasMember <https://solid.kjernsmo.net/profile/card#me>.
```

I don't know what the UID is, so I just put something in there. Isn't this about the minimal group listing there could be?

File: src/group-listing.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+   */
+  constructor (options = {}) {
+    this.uri = options.uri
+    this.uid = options.uid
+    this.members = options.members || new Set()
+    this.listing = options.listing
+    this.rdf = options.rdf
+    this.graph = options.graph
+    if (this.rdf && this.graph) {
+      this.initFromGraph(this.graph, this.rdf)
```
Comment (RubenVerborgh) [inline src/group-listing.js]: This can't work. `initFromGraph` expects `uri` as first param.

File: src/group-listing.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+   * @param [options.listing] {string|NamedNode} Group listing document URI
+   * @param [options.rdf] {RDF} RDF library
+   * @param [options.graph] {Graph} Parsed graph of the group listing document
+   */
+  constructor (options = {}) {
+    this.uri = options.uri
+    this.uid = options.uid
+    this.members = options.members || new Set()
+    this.listing = options.listing
+    this.rdf = options.rdf
```
Comment (RubenVerborgh) [inline src/group-listing.js]: should probably be `options.rdf || require('rdflib')`

File: src/group-listing.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    this.graph = options.graph
```
Comment (RubenVerborgh) [inline src/group-listing.js]: Should probably be `options.graph || this.rdf.graph()`

File: src/permission-set.js [hunk 4]
```diff
@@ -594,21 +704,24 @@ class PermissionSet {
...
       // Extract all 'Public' matches (agentClass foaf:Agent)
-      let publicMatches = graph.match(fragment, vocab.acl('agentClass'),
-        vocab.foaf('Agent'))
+      let publicMatches = graph.match(fragment, ns.acl('agentClass'),
+        ns.foaf('Agent'))
       // Extract all acl:agentGroup matches
-      let groupMatches = graph.match(fragment, vocab.acl('agentGroup'))
+      let groupMatches = graph.match(fragment, ns.acl('agentGroup'))
+      groupMatches = groupMatches.map(ea => {
+        return new GroupListing({ listing: ea })
```
Comment (RubenVerborgh) [inline src/permission-set.js]: This does not result in a parsed group (missing params).

File: src/group-listing.js (hunk 1 above)
Comment (RubenVerborgh) [reply src/group-listing.js]: …but we're not using this, I guess.

File: src/group-listing.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+   *   (e.g. `https://example.com/groups#management`)
+   * @param [options.uid] {string} Value of `vcard:hasUID` object
+   * @param [options.members={}] {Set} Set of group members, by webId
+   * @param [options.listing] {string|NamedNode} Group listing document URI
+   * @param [options.rdf] {RDF} RDF library
+   * @param [options.graph] {Graph} Parsed graph of the group listing document
+   */
+  constructor (options = {}) {
+    this.uri = options.uri
+    this.uid = options.uid
```
[the remark this answers is not in the document]
Comment (dmitrizagidulin) [reply src/group-listing.js]: Oh man, I do not remember...

**Pull request merged**